engine/compiler/script.go

```diff
@@ -27,7 +27,11 @@ func setupScript(src *resource.Step, dst *engine.Step, os string) {
 // helper function configures the pipeline script for the
 // windows operating system.
 func setupScriptWindows(src *resource.Step, dst *engine.Step) {
-	dst.Entrypoint = []string{"powershell", "-noprofile", "-noninteractive", "-command"}
+	sh := "powershell"
+	if len(src.Shell) != 0 {
+		sh = src.Shell
+	}
+	dst.Entrypoint = []string{sh, "-noprofile", "-noninteractive", "-command"}
 	dst.Command = []string{"echo $Env:DRONE_SCRIPT | iex"}
 	dst.Envs["DRONE_SCRIPT"] = powershell.Script(src.Commands)
 	dst.Envs["SHELL"] = "powershell.exe"
```

Comment:
Looking over the change once more, I realize that maybe we also want to pass the name of the overridden shell to the executing script. Although I am not sure about the role of the file extension here.

```suggestion
	dst.Envs["SHELL"] = sh
```

I am also not sure if/where a similar thing is done on the Linux side of things.